plan/config.go

```diff
@@ -12,41 +14,73 @@ const defaultCallTimeout = 5
 // ReadConfigFromEnviron creates a Config by looking for CROSSDOCK_ environment vars
 func ReadConfigFromEnviron() *Config {
 	const (
-		callTimeoutKey = "CROSSDOCK_CALL_TIMEOUT"
-		clientsKey     = "CROSSDOCK_CLIENTS"
-		axisKeyPrefix  = "CROSSDOCK_AXIS_"
+		callTimeoutKey    = "CROSSDOCK_CALL_TIMEOUT"
+		waitKey           = "CROSSDOCK_WAIT_FOR"
+		axisKeyPrefix     = "CROSSDOCK_AXIS_"
+		behaviorKeyPrefix = "CROSSDOCK_BEHAVIOR_"
 	)
 	callTimeout, _ := strconv.Atoi(os.Getenv(callTimeoutKey))
 	if callTimeout == 0 {
 		callTimeout = defaultCallTimeout
 	}
-	clients := trimCollection(strings.Split(os.Getenv(clientsKey), ","))
-	var axes []Axis
+	fmt.Println(os.Getenv(waitKey))
+	waiters := trimCollection(strings.Split(os.Getenv(waitKey), ","))
+	axes := make(map[string]Axis)
+	behaviors := make(map[string]Behavior)
 	for _, e := range os.Environ() {
-		if !strings.HasPrefix(e, axisKeyPrefix) {
-			continue
-		}
-		d := strings.TrimPrefix(e, axisKeyPrefix)
+		if strings.HasPrefix(e, axisKeyPrefix) {
+			d := strings.TrimPrefix(e, axisKeyPrefix)
+
+			pair := strings.SplitN(d, "=", 2)
+			key := strings.ToLower(pair[0])
+			values := strings.Split(pair[1], ",")
+			values = trimCollection(values)
+
+			axis := Axis{
+				Name:   key,
+				Values: values,
+			}
+			axes[key] = axis
+		} else if strings.HasPrefix(e, behaviorKeyPrefix) {
+			d := strings.TrimPrefix(e, behaviorKeyPrefix)
 
-		pair := strings.SplitN(d, "=", 2)
-		key := strings.ToLower(pair[0])
-		values := strings.Split(pair[1], ",")
-		values = trimCollection(values)
+			pair := strings.SplitN(d, "=", 2)
+			key := strings.ToLower(pair[0])
+			values := strings.Split(pair[1], ",")
+			values = trimCollection(values)
+			behavior := Behavior{
+				Name:    key,
+				Clients: values[0],
+				Params:  values[1:],
+			}
 
-		axis := Axis{
-			Name:   key,
-			Values: values,
+			behaviors[key] = behavior
 		}
-		axes = append(axes, axis)
 	}
 	config := &Config{
 		CallTimeout: time.Duration(callTimeout),
-		Clients:     clients,
+		Waiters:     waiters,
```

Comment:
Let's rename this to `WaitForHosts: hosts` - waiters makes me feel like those are the things that will be doing the waiting.